packages/client/components/BottomControlBarReadyButton.tsx

```diff
@@ -0,0 +1,169 @@
+  ({showShakeAnimation, delay}) => {
+    return {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      animation: showShakeAnimation ? `1s ease-in-out ${delay}s 3 ${shakeAnimation}` : undefined
+    }
+  }
+)
+
+interface Props {
+  meetingRef: BottomControlBarReadyButton_meeting$key
+  progress: number
+  children: ReactNode
+}
+
+const BottomControlBarReadyButton = (props: Props) => {
+  const {meetingRef, progress, children} = props
+  const meeting = useFragment(
+    graphql`
+      fragment BottomControlBarReadyButton_meeting on NewMeeting {
+        localPhase {
+          phaseType
+        }
+        ... on RetrospectiveMeeting {
+          endedAt
+          facilitatorUserId
+          localPhase {
+            ... on ReflectPhase {
+              reflectPrompts {
+                id
+                editorIds
+              }
+            }
+          }
+          reflectionGroups {
+            id
+            reflections {
+              id
+              isEditing
+              isDropping
+              isViewerDragging
+            }
+          }
+          viewerMeetingMember {
+            votesRemaining
+          }
+          votesRemaining
+        }
+      }
+    `,
+    meetingRef
+  )
+  const {
+    endedAt,
+    facilitatorUserId,
+    localPhase,
+    reflectionGroups,
+    viewerMeetingMember,
+    votesRemaining
+  } = meeting
+  const [delaySeconds, setDelaySeconds] = useState<number>()
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'reflect') return
+
+    const hasNoReflection = reflectionGroups?.every((group) => group.reflections?.length === 0)
+    if (hasNoReflection) return
+
+    const reflectPrompts = localPhase.reflectPrompts!
+    const beforeDemoStart =
+      isDemoRoute() && reflectPrompts.every(({editorIds}) => !Array.isArray(editorIds))
+    if (beforeDemoStart) return
+
+    const isNotEditing = reflectPrompts.every(({editorIds}) => {
+      return editorIds === undefined || (Array.isArray(editorIds) && editorIds.length === 0)
+    })
+
+    setDelaySeconds(isNotEditing ? 30 : undefined)
+  }, [localPhase.phaseType === 'reflect', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'group') return
+
+    const isNotDragging = reflectionGroups?.every((group) =>
+      group.reflections?.every(
+        (reflection) => !reflection.isViewerDragging && !reflection.isDropping
+      )
+    )
+
+    setDelaySeconds(isNotDragging ? 30 : undefined)
+  }, [localPhase.phaseType === 'group', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'vote') return
+
+    const teamVotesRemaining = votesRemaining || 0
+    const myVotesRemaining = viewerMeetingMember?.votesRemaining || 0
+    const isNotVoting = teamVotesRemaining === 0 || myVotesRemaining === 0
```

Comment:
-1 is there a situation where `myVotesRemaining > 0 && teamVotesRemaining === 0`? I think checking for `teamVotesRemaining` would be sufficient.
+1 `noVotesRemaining` would be a better variable name in my opinion.

Reply:
there's no situation where `myVotesRemaining > 0 && teamVotesRemaining === 0`? because `teamVotesRemaining` actually includes the count of `myVotesRemaining`, so I'll simplify the conditions here to:
```ts
const noVotesRemaining = teamVotesRemaining === 0
```

![Google Chrome-Retro #9  lv460051518+1’s Team-004077-20220823](https://user-images.githubusercontent.com/4997466/186153162-6d8feb6d-7c2b-477e-a9bd-596e1c43f440.png)

Reply:
The reason I used `myVotesRemaining === 0` before was that it's possible that not all team members were online and participated in the voting for this meeting. So it started prompting when all of the facilitator's own votes had been cast. But I'll address this kind of case in a separate issue.